sycl/doc/SharedLibraries.md

```diff
@@ -167,21 +190,99 @@ struct _pi_device_binary_property_struct {
 ```
 
 List of imported symbols is represented as a single property set with name
-`ImportedSymbols` recorded in the `Name` field of property set.
+`SYCL/imported symbols` recorded in the `Name` field of property set.
 Each property in this set holds name of the particular imported symbol recorded
 in the `Name` field of the property.
 List of exported symbols is represented in the same way, except the
-corresponding set has the name `ExportedSymbols`.
+corresponding set has the name `SYCL/exported symbols`.
 
 ### DPC++ runtime changes
 
 DPC++ RT performs *device images collection* task by grouping all device
 images required to execute a kernel based on the list of exports/imports and
 links them together using PI API.
 
-Given that all exports will be arranged to a separate device images without
-kernels it is reasonable to store device images with exports in a separate data
-structure.
+#### Device images collection
+
+DPC++ Runtime class named ProgramManager stores device images using following
+data structure:
+```
+/// Keeps all available device executable images added via \ref addImages.
+/// Organizes the images as a map from a kernel set id to the vector of images
+/// containing kernels from that set.
+/// Access must be guarded by the \ref Sync::getGlobalLock()
+std::unordered_map<SymbolSetId,
+                   std::unique_ptr<std::vector<RTDeviceBinaryImageUPtr>>>
+    m_DeviceImages;
+
+using StrToKSIdMap = std::unordered_map<string_class, SymbolSetId>;
+/// Maps names of kernels from a specific OS module (.exe .dll) to their set
+/// id (the sets are disjoint).
+std::unordered_map<OSModuleHandle, SymbolSetId> m_SymbolSets;
+```
+Assume each device image represents some combination of symbols and different
+device images may contain only exactly the same or not overlapping combination
+of symbols. If it is not so, there can be two cases:
+  - Symbols are the same. In this case it doesn't matter which device image is
+  taken to use duplicated symbol
+  - Symbols are not the same. In this case ODR violation takes place, such
```

Comment:
You really mean that symbols themselves are the same in both cases, but their definitions (or values) are not - right? Otherwise there is no duplication. Please clarify in the doc.

Reply:
Right, will clarify.